docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -526,6 +524,44 @@ public class MyConfiguration {
 ----
 ====
 
+===== Retry-aware instance selection
+
+For load-balanced retries, by default, we wrap the `ServiceInstanceListSupplier` bean with `RetryAwareServiceInstanceListSupplier` to select a different instance from the one previously chosen, if available. You can disable this behavior by setting the value of `spring.cloud.loadbalancer.retry.avoidPreviousInstance` to `false`.
+You can also instantiate `RetryAwareServiceInstanceListSupplier` on your own. Then, apart from the constructor that allows for passing only one delegate, you can also use `RetryAwareServiceInstanceListSupplier(ServiceInstanceListSupplier delegate, ServiceInstanceListSupplier secondaryDelegate)`. If you do, the secondary delegate will be used for providing instances during retries.
+
+You can use the builder we provide to easily instantiate suppliers, either with a secondary delegate:
+
+[source,java,indent=0]
+----
+	@Bean
+	ServiceInstanceListSupplier serviceInstanceListSupplier(ConfigurableApplicationContext context) {
+		return ServiceInstanceListSupplier
+				.builder()
+				.withDiscoveryClient()
+				.withRetryAwareness(ServiceInstanceListSupplier.builder()
+						.withDiscoveryClient()
+						.withHints()
+						.build(context))
+				.build(context);
+	}
+----
+
+or without it:
+
+[source,java,indent=0]
+----
+@Bean
+	ServiceInstanceListSupplier serviceInstanceListSupplier(ConfigurableApplicationContext context) {
+		return ServiceInstanceListSupplier
+				.builder()
+				.withDiscoveryClient()
+				.withRetryAwareness()
+				.build(context);
+	}
+----
+
+WARNING: If you opt for instantiating `RetryAwareServiceInstanceListSupplier` on your own, make sure you set the value of `spring.cloud.loadbalancer.retry.avoid-previous-instance` to `false` so that we do not instantiate one for you automatically.
```

Comment:
Why do we do anything if they supply their own bean?

Reply:
It would be better if they did not have to, however, the bean that we create automatically is created during the same configuration stage and `@ConditionalOnMissingBean` so, in order to avoid the programmatically created supplier getting wrapped by yet another retry supplier and at the same time, not change production behaviour, it could be set in this way. However, I don't like it a lot either, so if we can think of a better way, we could switch it up.